features/netsocket/EMACInterface.cpp

```diff
@@ -95,6 +95,15 @@ const char *EMACInterface::get_ip_address()
     return NULL;
 }
 
+nsapi_error_t EMACInterface::get_ipv6_link_local_address(SocketAddress *address)
+{
+    if (_interface && _interface->get_ipv6_link_local_address(address)) {
+        return NSAPI_ERROR_PARAMETER;
```

Comment:
You are translating all possible errors to `NSAPI_ERROR_PARAMETER`

Reply:
pushed code to return the return value of _interface->get_ipv6_link_local_address